cpp/tests/replace/replace_tests.cu

```diff
@@ -0,0 +1,143 @@
+         float chk[8] = { 99.0f };
+         cudaMemcpy(chk, mutable_view.head(), sizeof(float) * num_els, cudaMemcpyDeviceToHost);      
+         
+         // can't compare nans and -nans directly since they will always be equal, so we'll compare against
+         // bit patterns.       
+         uint32_t nan = hard_cast_float(NAN);            
+         EXPECT_TRUE(hard_cast_float(chk[3]) == nan);
+         EXPECT_TRUE(hard_cast_float(chk[4]) == nan);
+         EXPECT_TRUE(chk[1] == 0.0f);
+      }
+
+      // returned column overload
+      {
+         auto out = cudf::normalize_nans_and_zeros(test_data->view());
+
+         // get the data back
+         float chk[8] = { 99.0f };
+         cudaMemcpy(chk, out->view().head(), sizeof(float) * num_els, cudaMemcpyDeviceToHost);      
+         
+         // can't compare nans and -nans directly since they will always be equal, so we'll compare against
+         // bit patterns.       
+         uint32_t nan = hard_cast_float(NAN);            
+         EXPECT_TRUE(hard_cast_float(chk[3]) == nan);
+         EXPECT_TRUE(hard_cast_float(chk[4]) == nan);
+         EXPECT_TRUE(chk[1] == 0.0f);
+      }
+   }
+   
+    // doubles
+   {
+      double els[8] = { 32.5, -0.0, 111.0, -NAN, NAN, 1.0, 0.0, 54.3 }; 
+
+      // copy the data to mutable device column
+      auto test_data = cudf::make_numeric_column(cudf::data_type(cudf::FLOAT64), num_els, cudf::ALL_VALID, 0);
+      auto test_data_mview = test_data->mutable_view();
+      cudaMemcpy(test_data_mview.head(), els, sizeof(double) * num_els, cudaMemcpyHostToDevice);
+
+      // mutable overload
+      {                    
+         auto mutable_view = test_data->mutable_view(); 
+         cudf::normalize_nans_and_zeros(mutable_view);
+
+         // get the data back
+         double chk[8] = { 99.0 };
+         cudaMemcpy(chk, mutable_view.head(), sizeof(double) * num_els, cudaMemcpyDeviceToHost);      
+         
+         // can't compare nans and -nans directly since they will always be equal, so we'll compare against
+         // bit patterns.       
+         uint64_t nan = hard_cast_double(NAN);            
+         EXPECT_TRUE(hard_cast_double(chk[3]) == nan);
+         EXPECT_TRUE(hard_cast_double(chk[4]) == nan);
+         EXPECT_TRUE(chk[1] == 0.0);
+      }
+
+      // returned column overload
+      {
+         auto out = cudf::normalize_nans_and_zeros(test_data->view());
+
+         // get the data back
+         double chk[8] = { 99.0 };
+         cudaMemcpy(chk, out->view().head(), sizeof(double) * num_els, cudaMemcpyDeviceToHost);      
+         
+         // can't compare nans and -nans directly since they will always be equal, so we'll compare against
+         // bit patterns.       
+         uint64_t nan = hard_cast_double(NAN);            
+         EXPECT_TRUE(hard_cast_double(chk[3]) == nan);
+         EXPECT_TRUE(hard_cast_double(chk[4]) == nan);
+         EXPECT_TRUE(chk[1] == 0.0);
+      }
+   }
+}
```

Comment:
```suggestion
}

```

Reply:
Fixed in ff7f197e70fcccca7539a90eec5fade7f986e36f